src/Illuminate/Support/UriQueryString.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Illuminate\Support;
+
+use Illuminate\Contracts\Support\Arrayable;
+use Illuminate\Support\Traits\InteractsWithData;
+
+class UriQueryString implements Arrayable
+{
+    use InteractsWithData;
+
+    /**
+     * Create a new URI query string instance.
+     */
+    public function __construct(Uri $uri)
+    {
+        $this->uri = $uri;
+    }
```

Comment:
```suggestion
    public function __construct(protected Uri $uri)
    {
        //
    }
```

Avoid following deprecation:

```
2) /home/runner/work/framework/framework/src/Illuminate/Support/UriQueryString.php:17
Creation of dynamic property Illuminate\Support\UriQueryString::$uri is deprecated
```

Reply:
Can also use `public readonly Uri $uri` since the class only gets value from `$uri`.